lib/ansible/modules/web_infrastructure/ansible_tower/tower_inventory_source.py

```diff
@@ -238,11 +245,21 @@ def main():
             if module.params.get('description'):
                 params['description'] = module.params.get('description')
 
+            try: 
+                org_res = tower_cli.get_resource('organization')
+                org = org_res.get(name=organization)
+            except (exc.NotFound) as excinfo:
+                module.fail_json(
+                    msg='Failed to get organization,'
+                    'organization not found: {0}'.format(excinfo),
+                    changed=False
+                )
+
             if module.params.get('credential'):
                 credential_res = tower_cli.get_resource('credential')
                 try:
                     credential = credential_res.get(
-                        name=module.params.get('credential'))
+                        name=module.params.get('credential'), organization=org['id'])
```

Comment:
credential is particularly problematic because it may be owned by a single user, in which case the organization will be null. This is a common case, and wouldn't be possible by assuming the organization is shared as you are doing here.

Reply:
Ah ofcourse, didnt think of that one. 
Do you think a valid solution is to add all these as separated params, like: inventory_organization, credential_organization, source_project_organization, source_script_organization, and have them as 'optional' and not mandatory?

Reply:
This is a pain point shared among all Tower / AWX clients, and I wouldn't say that any of them have ever gotten it right. I don't like 2 separate parameters. In my perfect world, I think it would optionally take a string or a dict (and there would be a type option to allow this), and using the dict you could specify both the project name & organization. This is unlikely to happen.

The _primary_ fix you're getting in here, adding `organization` parameter to tower_inventory_source, and applying it to the inventory filter - that's really good. We need that. I would like to mostly focus from that and not worry too much about the other details which are still far from perfection.